internal/local/archive.go

```diff
@@ -19,143 +16,77 @@ const (
 	extZip   = ".zip"
 )
 
-var (
-	supportedExts = []string{extZip, extTar, extTgz, extTarGz}
-)
-
-// ArchiveReader provides sequential access to the contents of the supported archive formats
-type ArchiveReader interface {
-	Next() (FileHeader, error)
-	Read(b []byte) (int, error)
-}
-
-// FileHeader holds the path and file information
-type FileHeader struct {
-	Path string
-	Info os.FileInfo
-}
-
-func newArchiveReader(path string, file *os.File) (ArchiveReader, error) {
-	ext := strings.ToLower(filepath.Ext(path))
-
-	if ext == extZip {
-		return newZipReader(file)
-	}
-
-	if ext == extTar {
-		return newTarReader(file), nil
-	}
-
-	if ext == extTgz || strings.HasSuffix(path, extTarGz) {
-		return newGzipReader(file)
-	}
-
-	if ext == "" {
-		return newDirReader(path)
-	}
+type archiveMatcher func(path, ext string) bool
 
-	return nil, errUnknownArchiveExtension(path)
+func isZip(_, ext string) bool {
+	return ext == extZip
 }
 
-func errUnknownArchiveExtension(path string) error {
-	return fmt.Errorf(
-		"failed to read archive file at %s: unsupported format, use one of [%s] instead",
-		path,
-		strings.Join(supportedExts, ", "),
-	)
+func isTar(_, ext string) bool {
+	return ext == extTar
 }
 
-type zipReader struct {
-	*zip.Reader
-
-	currFileIdx  int
-	currOpenFile io.ReadCloser
+func isTarCompressed(path, ext string) bool {
+	return ext == extTgz || strings.HasSuffix(path, extTarGz)
 }
 
-func newZipReader(f *os.File) (ArchiveReader, error) {
-	fi, err := f.Stat()
-	if err != nil {
-		return nil, err
-	}
-
-	zr, err := zip.NewReader(f, fi.Size())
-	if err != nil {
-		return nil, err
+var (
+	preferredArchiveMatchers = []archiveMatcher{
+		isZip,
```

Comment:
open to suggestions here, but I check with andy and he thought we tend to read `.zip` files the fastest